crates/primitives/src/l1/proof.rs

```diff
@@ -0,0 +1,50 @@
+use arbitrary::Arbitrary;
+use bitcoin::Transaction;
+use borsh::{BorshDeserialize, BorshSerialize};
+use serde::{Deserialize, Serialize};
+
+use crate::buf::Buf32;
+
+/// A trait for computing some kind of transaction ID (e.g., [`Txid`](bitcoin::Txid) or
+/// [`Wtxid`](bitcoin::Wtxid)) from a [`Transaction`].
+///
+/// This trait is designed to be implemented by "marker" types that define how a transaction ID
+/// should be computed. For example, [`TxIdMarker`] invokes [`Transaction::compute_txid`], and
+/// [`WtxIdMarker`] invokes [`Transaction::compute_wtxid`]. This approach avoids duplicating
+/// inclusion-proof or serialization logic across multiple ID computations.
+pub trait TxIdComputable {
+    /// Computes the transaction ID for the given transaction.
+    ///
+    /// The `idx` parameter allows marker types to handle special cases such as the coinbase
+    /// transaction (which has a zero [`Wtxid`](bitcoin::Wtxid)) by looking up the transaction
+    /// index.
+    fn compute_id(tx: &Transaction, idx: usize) -> Buf32;
+}
+
+/// Marker type for computing the [`Txid`](bitcoin::Txid).
+#[derive(
+    Clone, Debug, PartialEq, Eq, Arbitrary, BorshSerialize, BorshDeserialize, Serialize, Deserialize,
+)]
+pub struct TxIdMarker;
+
+/// Marker type for computing the [`Wtxid`](bitcoin::Wtxid).
+#[derive(
+    Clone, Debug, PartialEq, Eq, Arbitrary, BorshSerialize, BorshDeserialize, Serialize, Deserialize,
+)]
+pub struct WtxIdMarker;
+
+impl TxIdComputable for TxIdMarker {
+    fn compute_id(tx: &Transaction, _idx: usize) -> Buf32 {
+        tx.compute_txid().into()
+    }
+}
+
+impl TxIdComputable for WtxIdMarker {
```

Comment:
What is the rationale for the word "marker" here?  "Selector" is probably more appropriate.

I don't see anywhere this is being used.  What purpose does this serve?  Was this included here as part of a dirty rebase?

Reply:
This is probably being used in Prover code.

> Was this included here as part of a dirty rebase?

I don't think so. This was here before, and is a part of this PR because I am splitting `primitives/l1.rs`.